Add GBV survey panel and update sexual misconduct pathways panel

wwidjaja1 opened the pull request:
## Description
_Adds the survey portion of gbv.  https://www.figma.com/design/1hkYSwY89cpDkj0Mej7iCA/GBV---for-Dev-team?node-id=9428-2115&p=f&t=vueJ710PvsGwOqQ2-0 Attempts to make use of survey library for state management where possible, but uses conditional logic in order to match the designers' intended flow for the questions and results that appear in the view. The PR itself may appear to just have a few line changes, though the whole SituationStepPanel.dart file should be reviewed along with the onNotSure function in SexualMisconductPathwaysPanel.dart. Remaining todos include

- The circular icon images within the text boxes have not been added yet.
- The Resources at the end show up as intended but don't yet have the red category text, e.g. "educational", "emergency."
- The back button may not work as expected after the last question is answered.
- Use Styles library where possible and match Figma colors and design elements
- Any refactoring required to follow existing conventions_

**Feature #(4954)**

## Review Time Estimate

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

mihail-varbanov reviewed (changes requested):
@wwidjaja1, please resolve all conflicts first before requesting a review & merge to [4954-gbv-master](https://github.com/rokwire/illinois-app/tree/4954-gbv-master). It is a good idea to merge [4954-gbv-master](https://github.com/rokwire/illinois-app/tree/4954-gbv-master) content first into your work branch before requesting a review, which will ensure your workbranch is up to date.

<img width="684" height="174" alt="Screenshot 2025-08-27 at 10 39 06" src="https://github.com/user-attachments/assets/7d58bdac-71fa-42c4-82fd-6da365041af3" />

mihail-varbanov reviewed (changes requested):
@wwidjaja1, I see a sea of ​​problems in your source code! I'll start with the simple ones that I've already talked about, and then I will move on to the more fundamental ones.

1\. I did not acknowledge any of my previous [recommendations](https://github.com/rokwire/illinois-app/pull/5290#issuecomment-3166967383). The statement that this is on your TODO list is not a reason for approval and merging to the [GBV master branch](https://github.com/rokwire/illinois-app/tree/4954-gbv-master). While some of the requests can be compromised and approved without being fulfilled (but should not include everything applicable), others are mandatory.

The mandatory requests are these about problems that could either crash the app (checking mounted state after async operations) or break UX standards (show progress indicator while async operations are executed).

Your source code currently sits as a patch in the application and does not comply with any internal standards. The code is poorly aligned, with inappropriate class names, does not acknowledge the  application's string and styles tables, and does not use the standard widgets that everyone else in the app uses (e.g. app header bar). If only one or two of the listed things were not fulfilled, a compromise could be made and it could be left for subsequent fulfillment, but when it comes to everything, such a compromise cannot be made. You need to realize that you are working within a heavy existing infrastructure, not doing a hello world demo project.

2\. There are other problems that I did not mention before. I am sure that you are familiar with most of them. This code cannot be approved and merged with the current Back button behavior. There are other design deviations that are good to be removed: add the icon of the more info section of a step page, content and layout of the survey result page (please see the images bellow).
<img width="437" height="350" alt="Screenshot 2025-08-29 at 10 05 00" src="https://github.com/user-attachments/assets/e6d999ef-c9f5-4c19-90d3-688aef5f35c3" />
<img width="788" height="788" alt="Screenshot 2025-08-29 at 10 20 00" src="https://github.com/user-attachments/assets/4a3a1951-5bba-4d05-a7c2-8cd574ce95ef" />

3\. The most fundamental problem in your implementation is your data model.

3\.1\. The general purpose of defining the survey content as external JSON data is so that it can be edited without requiring changes to the source code. In [_SituationStepPanelState._onOptionSelected](https://github.com/rokwire/illinois-app/blob/4954-gbv-survey/lib/ui/gbv/SituationStepPanel.dart#L46-L138) you have implemented flow logic that relies on a particular survey content. What about if the University staff decide to add one more step in the survey, update the steps flow logic, or update some of the result pages content? In general, you should not make any reference to particular steps or options from the survey, and should not construct result content in your code. This all should reside in the external JSON data.

3\.2\. If you remember, at the beginning of your work, I warned you that the Survey model was not suitable for what was required in the [GBV design](https://www.figma.com/design/1hkYSwY89cpDkj0Mej7iCA/GBV---for-Dev-team?node-id=9428-2115&p=f&t=5mu3rWwDCVMOsHeC-0). Currently you are not using any logic from the Surveys, you are only presenting part of your data in terms of the survey model and use its JSON serialization support. This doesn't help in any way and, on the contrary, it rather ties your hands to follow the survey model rather than what you really need for this task. This is why when I recommended using IDs instead of option titles, you answered that this wasn't in Stephen's model. Also, your data is currently scattered - one part is in the [survey definition](https://github.com/rokwire/illinois-app/blob/4954-gbv-survey/lib/ui/gbv/SexualMisconductPathwaysPanel.dart#L219), and another in the [rule JSON definition]([https://github.com/rokwire/illinois-app/blob/4954-gbv-survey/lib/service/GBVRules.dart#L](https://github.com/rokwire/illinois-app/blob/4954-gbv-survey/lib/service/GBVRules.dart#L6)). It is more consistent and reliable to collect this data together, because they are very strongly related to each other.

So, to summarize - as minimum you have to remove any logic and reference to particular data items from your source code, everything should be defined in an external JSON data. Good to have is to define your own data model that maps exactly what you need and that keeps all data together and consistent.

wwidjaja1 commented:
Thank you for the feedback Misho, I've made the following changes so far:
- Removed case logic from the .dart code so state is managed through the database
- Survey questions are all pulled from the database
- Survey logic is managed with the appropriate .evaluate() and .getFollowUp() functions
- Added mount checks to async functions
- UI for questions matches Figma design
- Back button works as intended and matches the rest of the app's UI
- Added _buildLoadingContent and _buildErrorContent functions

wwidjaja1 commented:
Thank you Misho. I appreciate this. For "1.4. When you get ready I will take care to obtain the survey ID ("cabb1338-48df-4299-8c2a-563e021f82ca") from app config. Please supply the corresponding survey IDs on the other environments (prod & test).", the ids are:

- Test: 3e823e2f-f881-41d2-9eed-ee0d6d682127z
- Prod: fab145d1-86d7-49cc-9666-20bda8713f19
